supporting/shimexe/shim.cs

```diff
@@ -91,10 +95,25 @@ internal struct PROCESS_INFORMATION {
             if(!string.IsNullOrEmpty(cmd_args)) cmd_args = " " + cmd_args;
             var cmd = "\"" + path + "\"" + cmd_args;
 
+            var environment_dictionary = Environment.GetEnvironmentVariables();
+            foreach (KeyValuePair<string, string> entry in config) {
+                if (entry.Key.StartsWith("env::")) {
+                    environment_dictionary[entry.Key.Substring(5)] = ExpandEnvironmentVariables(environment_dictionary, entry.Value);
```

Comment:
Wouldn't [`Environment.SetEnvironmentVariable()`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.setenvironmentvariable) work in this case?

Reply:
I guess it would. Should I move to it?